test/system/045-start.bats

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bats   -*- bats -*-
+
+load helpers
+
+@test "podman start --all - start all containers" {
+    run_podman run -d $IMAGE /bin/true
+    cid_none="$output"
+    run_podman run -d --restart=always $IMAGE /bin/true
```

Comment:
Hi @edsantiago - yes and no.
`run_podman run -d --restart=always $IMAGE /bin/true` should be exited less than a second and be caught by `podman start --all` which try to start it again.

Reply:
OK, the `--restart=always` container exits immediately. But it also restarts immediately, over and over and over. What will happen if `podman start --all` catches this container in the very very brief moment that it is in state Running? Will it restart it (and test will pass), or will it skip it, and test will fail because it doesn't see that CID? Based on my experiments with a `sleep 5` container (instead of `true`), I believe the latter will happen, therefore this test is racey. But I need confirmation from @containers/podman-maintainers .

Reply:
@edsantiago I see your point. Yeah I agree.

Reply:
The Libpod APIs return an error on starting a started container. The higher-level start code assumes that starting a started container is a no-op and so it squashes this error and returns cleanly. You're operating entirely in cmd/podman, which should be above the level of the squash - I imagine that any start on a running container will add it to the list of started containers, having taken no action.

Reply:
Thanks a lot @edsantiago - that's very helpful.